help_to_heat/frontdoor/views.py

```diff
@@ -301,18 +301,19 @@ def get_context(self, request, session_id, *args, **kwargs):
         return {"council_tax_band_options": council_tax_bands}
 
     def handle_post(self, request, session_id, page_name, data, is_change_page):
-        return super().handle_post(request, session_id, page_name, data, is_change_page)
+        session_data = interface.api.session.get_session(session_id)
+        uprn = session_data.get("uprn")
+        country = session_data.get("country")
+        epc = None
+        if uprn:
+            epc = interface.api.epc.get_epc(uprn, country)
+        return super().handle_post(request, session_id, page_name, data, is_change_page) if epc \
+            else redirect("frontdoor:page", session_id=session_id, page_name="benefits")
 
 
 @register_page("epc")
 class EpcView(PageView):
-    def get(self, request, session_id, page_name, data=None, errors=None, is_change_page=False):
-        if not errors:
-            errors = {}
-        data = {}
-        prev_page_url, next_page_url = get_prev_next_urls(session_id, page_name)
-        extra_context = self.get_context(request=request, session_id=session_id, page_name=page_name, data={})
-
+    def get_context(self, request, session_id, page_name, data):
```

Comment:
Everything here is just undoing the previous PR to get it back to how it was before, so that the page properly loads and makes the necessary API calls.